setup.py

```diff
@@ -507,9 +513,21 @@ def check_file(f):
     extra_compile_args += ['-DWITH_CUDNN']
 
 if WITH_NNPACK:
-    main_libraries += ['nnpack', 'pthreadpool']
-    include_dirs.extend(NNPACK_INCLUDE_DIRS)
-    library_dirs.append(NNPACK_LIB_DIR)
+    if NNPACK_IS_USER_DEFINED:
+        main_libraries += ['nnpack', 'pthreadpool']
+        include_dirs.extend(NNPACK_INCLUDE_DIRS)
+        library_dirs.append(NNPACK_LIB_DIR)
+    else:
```

Comment:
It seems to me that this logic should be in the nnpack tool. setup.py should be a simple scripts that only gathers information and adds it to the extension

Reply:
I disagree. I need this code logic to be in `setup.py` because it sets up where we install temporary files/headers etc. which `setup_helpers` do not know about. Otherwise, if we change `tmp_install` to `temp_install` or something, then NNPACK builds break and you have to look and manually change a second file to fix this. I'd rather have this patchwork duplication in two places.

Reply:
Oh I didn't notice that you copy the files to `tmp_install`! But why? Just add the dir where NNPACK headers are. We don't do this for any other dependency

Reply:
@apaszke - The headers are in the user's conda include directory. I discussed this with @ezyang and he suggested not to add this to `include_dirs` as then we are picking up all the other headers as well, which I guess can be problematic.

Reply:
Compilers look for headers in a well defined order. Everything should be fine as long as you add this dir at the end of include dirs.